airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/jdbc/AbstractJdbcSource.kt

```diff
@@ -270,9 +271,44 @@ abstract class AbstractJdbcSource<Datatype>(
                     CheckedFunction { connection: Connection -> connection.metaData.catalogs },
                     CheckedFunction { queryResult: ResultSet ->
                         sourceOperations.rowToJson(queryResult)
-                    }
+                    },
                 )
-            }
+            },
+            CheckedConsumer { database: JdbcDatabase ->
+                var schemas = ArrayList<String>()
+                if (config!!.has(JdbcUtils.SCHEMAS_KEY) && config[JdbcUtils.SCHEMAS_KEY].isArray) {
+                    for (schema in config[JdbcUtils.SCHEMAS_KEY]) {
+                        schemas.add(schema.asText())
+                    }
+                }
+                // if UI has schemas specified, check if the user has select access to any table
+                if (schemas.isNotEmpty()) {
+                    for (schema in schemas) {
+                        LOGGER.info {
+                            "Checking if the user can perform select to any table in schema: $schema"
+                        }
+                        val tablesOfSchema = database.metaData.getTables(null, schema, "%", null)
+                        if (tablesOfSchema.next()) {
+                            var privileges =
+                                getPrivilegesTableForCurrentUser<JdbcPrivilegeDto>(database, schema)
+                            if (privileges.isEmpty()) {
+                                LOGGER.info {
+                                    "No table from schema $schema is accessible for the user."
+                                }
+                                throw ConfigErrorException(
+                                    "User can not perform select to any table from schema: $schema"
+                                )
```

Comment:
suggestion: `"User lacks privileges to SELECT from any of the tables in schema '$schema'"`

Reply:
fixed.

Reply:
NIT: can we extract that function into it's own `overridable fun`? This way, it's easy to override that one only. I'm concerned that for some databases (specifically snowflake), those iterative metadata will be expensive.

Reply:
> NIT: can we extract that function into it's own `overridable fun`? This way, it's easy to override that one only. I'm concerned that for some databases (specifically snowflake), those iterative metadata will be expensive.

yes, that makes sense. Will make a change.

Reply:
> NIT: can we extract that function into it's own `overridable fun`? This way, it's easy to override that one only. I'm concerned that for some databases (specifically snowflake), those iterative metadata will be expensive.

@stephane-airbyte just pushed a change to implement this. Please take a look, thanks!